cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/common/AbstractWatcherThread.java

```diff
@@ -238,56 +233,48 @@ protected final ApiClient getApiClient() throws IOException {
   /**
    * Creates a new {@link Watchable} for watching for changes.
    *
-   * @param lastResourceVersion if provided, the watch will start from this version. If it is {@code null},
-   *                            a list of the latest resources will be fetch first to compute the difference with
-   *                            the previously known set and have the watch starting from there.
    * @return a {@link Watchable}
    */
-  private Watchable<DynamicKubernetesObject> createWatchable(@Nullable String lastResourceVersion)
-    throws IOException, ApiException {
+  private Watchable<DynamicKubernetesObject> createWatchable() throws IOException, ApiException {
 
     ApiClient apiClient = getApiClient();
     DynamicKubernetesApi api = new DynamicKubernetesApi(group, version, plural, apiClient);
 
     ListOptions options = new ListOptions();
     updateListOptions(options);
 
-    String resourceVersion = lastResourceVersion;
+    KubernetesApiResponse<DynamicKubernetesListObject> listResult = api.list(namespace, options);
 
-    if (resourceVersion == null) {
-      KubernetesApiResponse<DynamicKubernetesListObject> listResult = api.list(namespace, options);
+    // Throw exception if the list call failed
+    listResult.throwsApiException();
 
-      // Throw exception if the list call failed
-      listResult.throwsApiException();
+    DynamicKubernetesListObject listObject = listResult.getObject();
+    String resourceVersion = listObject.getMetadata().getResourceVersion();
```

Comment:
are we still using resource versions?

Reply:
Yes, at line 281 when we create the watch. It is needed since we want to see all changes after this version. This is the way to make sure we don't miss any events between the fetching of the list and the start of watching.